src/VirtoCommerce.MarketingModule.Data/Search/PromotionUsageSearchService.cs

```diff
@@ -51,17 +48,23 @@ public virtual async Task<PromotionUsageSearchResult> SearchUsagesAsync(Promotio
                                          .Skip(criteria.Skip).Take(criteria.Take)
                                          .ToArrayAsync();
                         result.Results = usages.Select(x => x.ToModel(AbstractTypeFactory<PromotionUsage>.TryCreateInstance())).ToList();
-                        foreach(var usage in result.Results)
-                        {
-                            cacheEntry.AddExpirationToken(PromotionUsageCacheRegion.CreateChangeToken(usage));
-                        }
                     }
+                    cacheEntry.AddExpirationToken(PromotionUsageCacheRegion.CreateChangeToken(new[] { criteria.PromotionId }));
+
                     return result;
                 }
             });
         }
 
-        protected  virtual IList<SortInfo> BuildSortExpression(PromotionUsageSearchCriteria criteria)
+        private static void EnsureAgrumentsValid(PromotionUsageSearchCriteria criteria)
```

Comment:
I think would be better to suppress or ignore sonarLint warning.

Reply:
Done